src/olympia/editors/models.py

```diff
@@ -78,7 +78,7 @@ def new_editors():
 
         return [dict(user=i.arguments[1],
                      created=i.created)
-                for i in items]
+                for i in items if i.arguments[1] in group.users.all()]
```

Comment:
I don't think we can solve this in a clean way - the `ActivityLog` model is just a mess - so your solution is fine. 

One thing you could do is load slightly more `ActivityLog` objects, then slice the list comprehension you're building so that there is never more than 5 items - Won't completely solve the issue (there will always be edge cases, depending on how lucky you are with the items you fetched), but it may help a little. I'm not sure this is worth it, depends on how rare and important you feel this bug is in the first place.